src/spike/index.test.js

```diff
@@ -38,7 +38,48 @@ test('createSignedRequest', async ({ same }) => {
       Date: 1602872645,
       'Content-Type': 'application/json',
       Digest: 'sha-512=+PtokCNHosgo04ww4cNhd4yJxhMjLzWjDAKtKwQZDT4Ef9v/PrS/+BQLX4IX5dZkUMK/tQo7Uyc68RkhNyCZVg==',
-      Signature: 'keyId="test-key-a", algorithm="hs2019", headers="(request-target) (created) (expires) host digest content-type", signature="HFIn7RU3zvfFHeyOjsRlMzUQ18prW+KE61ikKKKoyAlcxGUdlBF/sruA+VznOhQNlWh3J4y3tZc8aDa0TxRBEg=="'
+      Signature: 'keyId="test-key-a", algorithm="hs2019", headers="(request-target) (created) (expires) host digest content-type", signature="HFIn7RU3zvfFHeyOjsRlMzUQ18prW+KE61ikKKKoyAlcxGUdlBF/sruA+VznOhQNlWh3J4y3tZc8aDa0TxRBEg==", created="1602872645", expires="1602872945"'
+    }
+  }
+  same(actual, expected)
+})
+
+test('createSignedRequest - no (created) field', async ({ same }) => {
+  const signatureFields = [
+    '(request-target)',
+    '(expires)',
+    'Host',
+    'Digest',
+    'Content-Type'
+  ]
+
+  const opts = {
+    secret: 'topSecret',
+    keyId: 'test-key-a',
+    signatureFields
+  }
+
+  const request = {
+    url: 'http://localhost:3000/foo',
+    method: 'POST',
+    body: JSON.stringify({ hello: 'world' }),
+    headers: {
+      Date: 1602872645,
```

Comment:
We might need to look at the way the extractors are set up for `(created)` and `(expires)`.

There are two issues that I see.
1. The [IEFT spec ](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-message-signatures-01#section-2.4) defines the created (and by extension expires) fields as "number of seconds since
   the Epoch". i.e. `parseInt(Date.now()/1000) // => 1614043967`
2. The HTTP Date Header is [defined](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Date) as a "forbidden header" and is formatted as a UTC date string e.g. `Date: Wed, 21 Oct 2015 07:28:00 GMT` 

 `extractors.js` defines the default extractors so:
```js
  '(created)': (req) => parseInt(req.headers.Date),
  '(expires)': (req) => parseInt(req.headers.Date) + EXPIRATION_OFFSET,
```

While the current implementation will work, I'm not sure it conforms with the IEFT spec ... It also requires the user to add a `Date` header to the proto-request object before having it signed. Would it be more ergonomic to determine and set the created value within the extractor function?

I see two options:

Option 1:
- Users need to pass in the reference for the (created) field with a `Date` header in the request passed to `signRequest()`
  - we need to handle converting from a UTC date string => milliseconds since Epoch => seconds since Epoch.
  - I'm also not sure without more research what "forbidden header" means in this context ... maybe we shouldn't be using this header at all?
 
Option 2:
- Change the "extractor" for `(created)` to actually set the created value using `parseInt(Date.now()/1000)` and not require a Date header value to be passed in with the request. 
- This makes testing slightly more difficult since `Date.now()` needs to be [mocked for repeatable testing](https://github.com/autotelic/ietf-hmac-signatures/blob/main/src/createSignRequest.test.js#L28-L35).
- This is the approach I took with the [current release](https://github.com/autotelic/ietf-hmac-signatures/blob/main/src/createSignRequest.js#L43)

Reply:
This is definitely a good point to raise. I think it's out of scope for this PR, but can you create an issue for it?  We can address it in another PR.

Reply:
No problem! I created issue #15 so we can discuss this comment. I'll approve the work in this PR.